### ISSUE TITLE
cannot run file in terminal --- Neon3 and Ubuntu 16.04

### ISSUE THREAD

**Issue by dm1seven** [anb0s/EasyShell#141]
The menu item Run with Gnome Terminal does not show anything. I think the command line for this item is not correct.  I tried the command line bellow: it works if I directly call it in gnome terminal with the actual file names,  but still does not work from EasyShell in Eclipse.

gnome-terminal --working-directory=${easyshell:container_loc} -e "bash -c ./${easyshell:resource_name}; bash"

Source of the custom command:
http://stackoverflow.com/questions/38135995/how-to-open-a-gnome-terminal-to-execute-a-command-with-gnome-terminal-constantl

**Comment by anb0s** [comment]
Thanks for reporting. I see two possible problems and solutions:
- see #124: some commands are broken in 2.0, please verify the testing build of v2.1 from #135. After upgrading the plugin, check if the command is different. If not, then remove your settings from this command: "EasyShell -> (2) Command -> Gnome Terminal -> Remove"
- see #6: may be the working directory command-line parameter is not working here too, but it was fixed with #66 and now the directory is set before executing the line from EasyShell, please verify this setting or provide screenshot from the command edit dialog

**Comment by anb0s** [comment]
If it still not working, please enable the debug output:
![image](https://cloud.githubusercontent.com/assets/95811/25120828/dc1588a4-241f-11e7-9df7-b2983c005e00.png)
and send me the output after executing the command...

**Comment by MariusDrulea** [comment]
I have tested the 2.1 build. I have the same behavior as before. Here are the debug messages:
![image](https://cloud.githubusercontent.com/assets/27725572/25126134/e0fbc9fe-2439-11e7-8731-e6234780a29f.png)

**Comment by anb0s** [comment]
Thanks. The log looks good for me. Please check if this helps:
- verify if gnome-terminal has a bug described in issue #6:  enable this checkbox in the command: 
![image](https://cloud.githubusercontent.com/assets/95811/25128156/f0de54fa-2438-11e7-806f-7d65c7ff47a0.png)
  it will set working dir before starting gnome-terminal, so this can be removed from command: "--working-directory=${easyshell:container_loc}"
- remove single quotes from command:
`  gnome-terminal --working-directory=${easyshell:container_loc} --command=./"${easyshell:resource_name}"
`
I've already done it, but will verify it again today at home with my Debian 9.0 system...

**Comment by MariusDrulea** [comment]
I have found the command actually got executed, but nothing is displayed because the terminal closes quickly after the bash script is executed. 

I have added: int n; cin >> n; into the code, to wait for the user command and the terminal appeared after that.

**Comment by MariusDrulea** [comment]
In order to keep the terminal open without `cin >> n`, I have used the following command pattern:
`gnome-terminal --working-directory=${easyshell:container_loc} -e "bash -c ./'${easyshell:resource_name}';bash"`.

I have found this option at this [link](http://stackoverflow.com/questions/38135995/how-to-open-a-gnome-terminal-to-execute-a-command-with-gnome-terminal-constantl)
The terminal stays alive with this command.

However, I don't fully like it because when I close it I get the following warning, see the picture.
![image](https://cloud.githubusercontent.com/assets/27725572/25137633/06cdbac6-2461-11e7-8881-3851e739ec69.png)

**Comment by anb0s** [comment]
Here are the parameters: http://manpages.ubuntu.com/manpages/xenial/man1/gnome-terminal.1.html
I've tried some variants and see no other solution now, like you described:
`gnome-terminal --working-directory=${easyshell:container_loc} -e "bash -c ./'${easyshell:resource_name}';bash"
`
Thanks!

Alternatives:
I've had always problems with using gnome-terminal and use xfce-terminal or sakura that have --hold parameter for this:
`xfce4-terminal --working-directory=${easyshell:container_loc} --command=./"'${easyshell:resource_name}'" --hold`

`sakura --execute=./"'${easyshell:resource_name}'" --hold
`

**Comment by anb0s** [comment]
With Debian 9.0 i cannot execute with the original gnome-command, only with -e "bash -c..." it works!
@MariusDrulea: does the original command from EasyShell works for you with `int n; cin >> n;` at Ubuntu ?

**Comment by anb0s** [comment]
Another possibility to hold the gnome-terminal is to call a script with or similar:
```
#!/bin/sh
'$1'
read -p "Press space to continue..." var_name
```
I've already some commands that uses this way...

**Comment by MariusDrulea** [comment]
`int n; cin >> n; ` works in Ubuntu with the original gnome-terminal command, see it bellow. The terminal waits for the user to enter the number.
`gnome-terminal --working-directory=${easyshell:container_loc} --command=./"'${easyshell:resource_name}'"`

**Comment by MariusDrulea** [comment]
I have finally found a solution. The command bellow works as expected. The trick was to add `exec bash` at the end, not just `bash`. 
:+1: 
`gnome-terminal --working-directory=${easyshell:container_loc} -e "bash -c ./'${easyshell:resource_name}';'exec bash'"`

The explanation is bellow and at the following link: [http://stackoverflow.com/questions/3512055/avoid-gnome-terminal-close-after-script-execution](http://stackoverflow.com/questions/3512055/avoid-gnome-terminal-close-after-script-execution)

> `exec` causes the running process to be replaced by the new process, otherwise you will have two bash processes running

Since there is a single bash process running in the terminal, the "Close this terminal?" warning no longer appears.

**Comment by anb0s** [comment]
Cool :) i will add this command as "Run Gnome Terminal (noclose or hold)" for next release too. Thanks!

**Comment by anb0s** [comment]
will be done with #144
